cgroup/cgroup_collector.go

```diff
@@ -0,0 +1,124 @@
+package cgroup
+
+import (
+	"bufio"
+	"github.com/prometheus/client_golang/prometheus"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type cgroupMetric struct {
+	Metric *prometheus.Desc
+	Path   string
+}
+
+type statMetric struct {
+	Metric     *prometheus.Desc
+	NameInStat string
+}
+
+type cgroupMemoryCollector struct {
+	metrics     []cgroupMetric
+	statMetrics []statMetric
+}
+
+func NewCGroupMemoryCollector() prometheus.Collector {
+	// Initialize all desired metrics with their descriptions
+	metrics := []cgroupMetric{
+		{
+			Path:   "/sys/fs/cgroup/memory/memory.kmem.usage_in_bytes",
+			Metric: prometheus.NewDesc("cgroup_kmem_memory_usage_bytes", "Memory usage of the cgroup in bytes.", nil, nil),
+		},
+		{
+			Path:   "/sys/fs/cgroup/memory/memory.kmem.max_usage_in_bytes",
+			Metric: prometheus.NewDesc("cgroup_kmem_memory_max_usage_bytes", "Maximum memory usage of the cgroup in bytes.", nil, nil),
+		},
+		{
+			Path:   "/sys/fs/cgroup/memory/memory.kmem.limit_in_bytes",
+			Metric: prometheus.NewDesc("cgroup_kmem_memory_limit_bytes", "Memory limit of the cgroup in bytes.", nil, nil),
+		},
+		{
+			Path:   "/sys/fs/cgroup/memory/memory.kmem.failcnt",
+			Metric: prometheus.NewDesc("cgroup_kmem_memory_failcnt", "Number of memory usage hits limits.", nil, nil),
```

Comment:
For `memory.kmem`, let's update the documentation line to say "kernel memory" (prom and grafana display these hints, so we should make them as accurate as possible"

For `memory.kmem.tcp`, it's the same for "TCP buffer memory (allocations|usage|hits)"